jerry-core/ecma/base/ecma-helpers-external-pointers.c

```diff
@@ -64,28 +64,29 @@ ecma_create_external_pointer_property (ecma_object_t *obj_p, /**< object to crea
   JERRY_STATIC_ASSERT (sizeof (uint32_t) <= sizeof (ECMA_PROPERTY_VALUE_PTR (prop_p)->value),
                        size_of_internal_property_value_must_be_greater_than_or_equal_to_4_bytes);
 
-  if (sizeof (ecma_external_pointer_t) == sizeof (uint32_t))
+#ifdef ECMA_VALUE_CAN_STORE_UINTPTR_VALUE_DIRECTLY
+
+  ECMA_PROPERTY_VALUE_PTR (prop_p)->value = (uint32_t) ptr_value;
```

Comment:
I think ecma_value_t would be a better (future proof) for casting instead of uint32_t.